src/main/java/com/broadleafcommerce/autoconfigure/HSQLDBServer.java

```diff
@@ -64,20 +68,62 @@ public HSQLDBServer(final HSQLDBProperties autoProps, Environment environment) {
 
     @Override
     public boolean isRunning() {
-        try (Socket ignored = new Socket(InetAddress.getByName(null), props.getIntegerProperty("server.port", 0))) {
-            return true;
-        } catch (IOException ignored) {
-            return false;
+        boolean isRunning = false;
+        Connection c = null;
+
+        final int port = props.getIntegerProperty("server.port", 0);
+        try {
+            Class.forName("org.hsqldb.jdbc.JDBCDriver");
+            final String url = "jdbc:hsqldb:hsql://" + InetAddress.getByName(null).getHostName() + ":"
```

Comment:
I am not sure how reliable `InetAddress.getByName()` actually is. There are instances where you might not actually have this host in `/etc/hosts` and this particular call causes the entire startup to hang for basically forever (or like 5 minutes). I have personally seen this happen on OSX when I did not have the IPV6 entry for `localhost` in my `/etc/hosts`.

On VMs on various cloud providers I have also seen entries in `/etc/hosts` that map the host name to the _external_ IP address which means all traffic has to go out and back in again... not great.

Since this whole project assumes that HSQL and everything else is local, we should just use `127.0.0.1` since there is no OS-level translation that has to go on with host names.